integration/entity.go

```diff
@@ -0,0 +1,91 @@
+package integration
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/newrelic/infra-integrations-sdk/metric"
+	"github.com/newrelic/infra-integrations-sdk/persist"
+	"github.com/pkg/errors"
+)
+
+// Entity is the producer of the data. Entity could be a host, a container, a pod, or whatever unit of meaning.
+type Entity struct {
+	lock      sync.Mutex
+	Metadata  EntityMetadata   `json:"entity"`
+	Metrics   []*metric.Set    `json:"metrics"`
+	Inventory metric.Inventory `json:"inventory"`
+	Events    []*metric.Event  `json:"events"`
+	storer    persist.Storer
+}
+
+// EntityMetadata stores entity Metadata
+type EntityMetadata struct {
+	Name string `json:"name"`
+	Type string `json:"type""`
+}
+
+// EntityID entity identifier
+type EntityID string
+
+// NewEntity creates a new remote-entity.
+func NewEntity(entityName, entityType string) (*Entity, error) {
```

Comment:
*NOT FOR DOING IN THIS PULL REQUEST, BUT TO HAVE IT IN MIND*

Since an `Entity` needs a `Storer`, I wonder whether it would make sense to make `Integration` as the receiver of this function, so `Integration` automatically adds the `Storer`.

Also, we will need to add synchronization to `Metrics` and `Events`, so if this function is part of `Integration`, we could add a `Lock` only if the `Integration` is synchronized.

Reply:
Added